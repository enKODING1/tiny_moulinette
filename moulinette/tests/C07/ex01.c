#include <stdio.h>

int *ft_range(int min, int max);

void test(int min, int max)
{
    int *arr = ft_range(min, max);
    if (min >= max)
    {
        if (arr != NULL)
        {
            printf("Expected NULL, but not NULL.\n");
            return;
        }
        else
        {
            printf("Expected NULL, got NULL.\n");
            return;
        }
    }
    for (int i = 0; i < max - min; i++)
    {
        printf("%d ", arr[i]);
    }
    printf("\n");
}

int main(void)
{
    test(0, 0);
    test(1, 0);
    test(-1, -1);
    test(1, -1);
    test(-1, 0);
    test(-1, 1);
    test(4, 2);
    test(30, 40);
    return 0;
}
