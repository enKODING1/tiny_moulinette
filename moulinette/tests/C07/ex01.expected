Expected NULL, got NULL.
Expected NULL, got NULL.
Expected NULL, got NULL.
Expected NULL, got NULL.
-1 
-1 0 
Expected NULL, got NULL.
30 31 32 33 34 35 36 37 38 39 
